### PR TITLE
My branch

### PR DESCRIPTION
**Pull request by NagasairamM**
please verify the code and update in the main branch.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -12,86 +12,37 @@ provider "snowflake" {
...
+  parent_role_name = "SYSADMIN"
 }
-
-resource "snowflake_grant_privileges_to_role" "schema_grant" {
-  provider   = snowflake.security_admin
-  privileges = ["USAGE"]
-  role_name  = snowflake_role.role.name
+resource "snowflake_grant_privileges_to_account_role" "schema_grant" {
+  provider          = snowflake.security_admin
+  privileges        = ["USAGE"]
```
Comment (csarat) [inline main.tf:42]: Why did we give `usage` privileges here?

File: main.tf [hunk 2]
```diff
@@ -12,86 +12,37 @@ provider "snowflake" {
...
+  account_role_name = snowflake_role.role.name
   on_schema {
     schema_name = "\"${snowflake_database.db.name}\".\"${snowflake_schema.schema.name}\""
```
Comment (csarat) [inline main.tf:45]: Why do we have to escape quotes here? Why can't we directly reference the resource instance?

**Review by csarat — CHANGES_REQUESTED** [review]
Left comments

File: main.tf (hunk 2 above)
Comment (NagasairamM) [reply main.tf:45]: Actually, I tried that. it showing a syntax error. so bought it from documentation.

**Review by NagasairamM — COMMENTED** [review]
Please review the code.

File: main.tf (hunk 1 above)
Comment (csarat) [reply main.tf]: questions should be responded here and should resolve the conversation here

**Review by csarat — APPROVED** [review]
LGTM

**Pull request merged**